src/java.base/share/classes/java/util/Properties.java

```diff
@@ -1515,7 +1515,7 @@ public synchronized Object computeIfAbsent(Object key,
 
     @Override
     public synchronized @Nullable Object compute(Object key,
-            BiFunction<? super Object, ?, ?> remappingFunction) {
+            BiFunction<? super Object, ? super Object, ? extends @Nullable Object> remappingFunction) {
```

Comment:
- Is the change from `?` to `? extends @Nullable Object` (for the `BiFunction` output) necessary for technical reasons? I would expect that we'd want to stick with `?` to match [what the JDK has](https://github.com/openjdk/jdk/blob/a5503fb6fa0ad8ae7472a1c849723357da581986/src/java.base/share/classes/java/util/Properties.java#L1507): With the annotation on the wildcard bound, which doesn't exist in the JDK class file, we end up with `1: #8(): METHOD_FORMAL_PARAMETER, param_index=1, location=[TYPE_ARGUMENT(2), WILDCARD]`, i.e., an annotation for a bound that doesn't exist. (We had trouble with bogus annotation locations previously in https://github.com/jspecify/jdk/pull/56.) Now maybe the JDK has had different types at different times (though I checked JDK 11 and JDK 17, and both had the bare `?`).
- The change from `?` to `? super Object` (for the second input of the `BiFunction`) seems strange in another way: That parameter is normally nullable (including in `Map` and in `Hashtable`). I could understand `? super @Nullable Object` if we needed that instead of `?` (as discussed in the previous bullet), but the non-null type is stranger.

If it turns out that we do need the types written this way here and need them written differently inside Google, we can accomplish it with a patch easily enough. I just want to understand where we stand (and be ready to warn NullAway or others if we learn of problems).

Reply:
The original signature produces a Java error:

````
src/java.base/share/classes/java/util/Properties.java:1517: error: name clash: compute(Object,BiFunction<? super Object,?,?>) in Properties and compute(Object,BiFunction<? super Object,? super Object,? extends Object>) in Hashtable have the same erasure, yet neither overrides the other
    public synchronized @Nullable Object compute(Object key,
                                         ^
````

I need to change it to at least `BiFunction<? super Object,? super Object,?>` to make Java happy. I removed the `@Nullable`, I added that too eagerly.
Is the new signature okay?